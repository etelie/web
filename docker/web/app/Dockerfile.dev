##### DEPENDENCIES

FROM arm64v8/node:19-alpine3.16 AS deps

# RUN apk add --no-cache libc6-compat openssl1.1-compat

WORKDIR /app

# Install Prisma Client
COPY ../../../prisma ./

# Install dependencies based on the preferred package manager
COPY ../../../package.json       ./
COPY ../../../package-lock.json* ./
RUN if [ -f package-lock.json ]; then npm ci; \
    else echo "Lockfile not found." && exit 1; \
    fi

##### BUILDER

FROM arm64v8/node:19-alpine3.16 AS builder
ARG DATABASE_URL
WORKDIR /app
COPY --from=deps /app/node_modules ./node_modules
COPY . .

RUN \
    if [ -f package-lock.json ]; then npm run build; \
    else echo "Lockfile not found." && exit 1; \
    fi

##### RUNNER

FROM arm64v8/node:19-alpine3.16 AS runner
WORKDIR /app

ENV NODE_ENV production

RUN addgroup --system --gid 1001 nodejs
RUN adduser --system --uid 1001 nextjs

COPY --from=builder /app/next.config.mjs    ./
COPY --from=builder /app/public             ./public
COPY --from=builder /app/package.json       ./package.json

COPY --from=builder --chown=nextjs:nodejs /app/.next/standalone ./
COPY --from=builder --chown=nextjs:nodejs /app/.next/static     ./.next/static

USER nextjs
EXPOSE 3000
ENV PORT 3000

CMD ["node", "server.js"]
