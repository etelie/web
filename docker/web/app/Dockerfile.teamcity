##### Build Docker image from Vite project outputs

FROM arm64v8/node:19-alpine3.16
WORKDIR /web

ENV NODE_ENV production
ENV PORT 3000

RUN addgroup --system --gid 1001 etelie
RUN adduser --system --uid 1001 web

COPY --chown=etelie:web ../../dist ./dist

USER web:etelie
EXPOSE ${PORT}

CMD ["node", "server.js"]
