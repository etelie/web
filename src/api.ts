import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import {
  BaseQueryFn,
  FetchArgs,
  FetchBaseQueryError,
  FetchBaseQueryMeta,
} from '@reduxjs/toolkit/query';
import { EndpointBuilder as _EndpointBuilder } from '@reduxjs/toolkit/dist/query/endpointDefinitions';

import { newsletterEndpoints } from '@/common/newsletter/endpoints';

const server = {
  protocol: import.meta.env.PROD ? 'https' : 'http',
  host: import.meta.env.PROD ? 'etelie.com' : 'localhost',
  port: import.meta.env.PROD ? 443 : 402,
};

export const apiSlice = createApi({
  reducerPath: 'api',
  baseQuery: fetchBaseQuery({
    baseUrl: `${server.protocol}://${server.host}:${server.port}`,
    credentials: 'same-origin',
    headers: {
      'content-type': 'application/json',
    },
    prepareHeaders: headers => {},
  }),
  endpoints: builder => ({
    ...newsletterEndpoints(builder),
  }),
});

export const mockApiSlice = createApi({
  reducerPath: 'api',
  baseQuery: fetchBaseQuery({
    baseUrl: '',
    credentials: 'same-origin',
    headers: {
      'content-type': 'application/json',
    },
    prepareHeaders: headers => {},
  }),
  endpoints: builder => ({}),
});

export type EndpointBuilder = _EndpointBuilder<
  BaseQueryFn<string | FetchArgs, unknown, FetchBaseQueryError, {}, FetchBaseQueryMeta>,
  never,
  'api'
>;
