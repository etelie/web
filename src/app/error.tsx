"use client"

const RootError = () => <div>Error loading the Etelie web application</div>;

export default RootError;
