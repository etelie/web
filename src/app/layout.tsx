const RootLayout = ({ children }: { children: React.ReactNode }) => (
  <html lang="en">
    <body>{children}</body>
  </html>
);

/**
 * Manage <head> HTML elements
 */
export const metadata = {
  title: "etelie",
};

export default RootLayout;
