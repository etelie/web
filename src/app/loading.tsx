const RootLoading = () => (
    <div>Loading etelie web application...</div>
)

export default RootLoading;
