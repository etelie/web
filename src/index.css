@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  /* Common shadow style for foreground elements (TitleCard, Modal, etc.) */
  .e-shadow-fore {
    @apply tw-shadow-lg tw-shadow-slate-300;
  }
}
