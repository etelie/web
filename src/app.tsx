import { renderTimestamp } from '@/common/time';

import { Modal, TitleCard } from '@/components/layout';
import { EmailForm } from '@/components/layout/EmailForm';
import { ModalControlOptions } from '@/components/layout/Modal/ModalControl';
import { useAppSelector } from '@/store';

export default () => {
  const locale = useAppSelector(state => state.locale.language);
  const timestamp = renderTimestamp(new Date(), locale);

  return (
    <>
      <TitleCard />
      <Modal
        hidden={false}
        control={ModalControlOptions.RELOAD}
        title={'etelie.com is under development'}
        footerLeft='Please check back in later'
        footerRight={timestamp}
      >
        {false && <EmailForm />}
      </Modal>
    </>
  );
};
