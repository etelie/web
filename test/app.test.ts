import { assert, describe, expect, test } from "vitest";

import { app_util } from "../src/app";

describe("App component", () => {
  test("assert", () => {
    assert(true);
  });

  test("expect", () => {
    expect(true).toBe(true);
  });

  test("snapshot", () => {
    expect(app_util()).toMatchSnapshot();
    expect(app_util()).toMatchInlineSnapshot('"x"');
  });
});
