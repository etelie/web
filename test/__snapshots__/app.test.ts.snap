// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > snapshot 1`] = `"x"`;
